python/lsst/pipe/tasks/processFakes.py

```diff
@@ -0,0 +1,316 @@
+        Adds an ``x`` and ``y`` column to the catalog of fake sources.
+    `trimFakeCat`
+        Trim the fake cat to about the size of the input image.
+    `mkFakeGalsimGalaxies`
+        Use Galsim to make fake double sersic galaxies for each set of galaxy parameters in the input file.
+    `mkFakeStars`
+        Use the PSF information from the calexp to make a fake star using the magnitude information from the
+        input file.
+    `cleanCat`
+        Remove rows of the input fake catalog which have half light radius, of either the bulge or the disk,
+        that are 0.
+    `addFakeSources`
+        Add the fake sources to the calexp.
+
+    Notes
+    -----
+    The ``calexp`` with fake souces added to it is written out as the datatype ``calexp_fakes``.
+    """
+
+    _DefaultName = "processCcdWithFakes"
+    ConfigClass = ProcessCcdWithFakesConfig
+
+    def __init__(self, schema=None, **kwargs):
+        """Initalize tings! This should go above in the class docstring
+        """
+
+        super().__init__(**kwargs)
+
+        if schema is None:
+            schema = SourceTable.makeMinimalSchema()
+        self.schema = schema
+        self.makeSubtask("insertFakes")
+        self.algMetadata = dafBase.PropertyList()
+        self.makeSubtask("detection", schema=self.schema)
+        self.makeSubtask("deblend", schema=self.schema)
+        self.makeSubtask("measurement", schema=self.schema, algMetadata=self.algMetadata)
+        self.makeSubtask("applyApCorr", schema=self.schema)
+        self.makeSubtask("catalogCalculation", schema=self.schema)
+
+    def runDataRef(self, dataRef):
+        """Read in/write out the required data products and add fake sources to the calexp.
+
+        Parameters
+        ----------
+        dataRef : `lsst.daf.persistence.butlerSubset.ButlerDataRef`
+            Data reference defining the ccd to have fakes added to it.
+            Used to access the following data products:
+                calexp
+                jointcal_wcs
+                jointcal_photoCalib
+
+        Notes
+        -----
+        Uses the calibration and WCS information attached to the calexp for the posistioning and calibration
+        of the sources unless the config option config.useUpdatedCalibs is set then it uses the
+        meas_mosaic/jointCal outputs. The config defualts for the column names in the catalog of fakes are
+        taken from the University of Washington simulations database. Operates on one ccd at a time.
+        """
+        exposureIdInfo = dataRef.get("expIdInfo")
+
+        if self.config.insertFakes.fakeType == "snapshot":
+            fakeCat = dataRef.get("fakeSourceCat").toDataFrame()
+        elif self.config.insertFakes.fakeType == "static":
+            fakeCat = dataRef.get("deepCoadd_fakeSourceCat").toDataFrame()
+        else:
+            fakeCat = Table.read(self.config.insertFakes.fakeType).to_pandas()
```

Comment:
Same comment as in `insertFakes.py`.  Should we be reading `self.config.insertFakes.fakeCat`?